spec/system/admin/admin_visits_overview_spec.rb

```diff
@@ -1,14 +1,54 @@
 require "rails_helper"
 
 RSpec.describe "Admin visits overview page", type: :system, js: true do
+  include ActiveSupport::Testing::TimeHelpers
+
   let(:super_admin) { create(:user, :super_admin) }
 
   before do
     sign_in super_admin
     visit admin_path
   end
 
-  it "has analytics" do
-    expect(page.body).to include "Analytics and trends"
+  describe "analytics" do
+    before do
+      travel_to Time.zone.local(2022, 3, 30)
+    end
+
+    it "has analytics" do
+      expect(page.body).to include "Analytics and trends"
+    end
+
+    it "starts with yesterday" do
+      expect(page.body).to include "Yesterday"
+    end
+
+    it "starts a week before" do
+      expect(page.body).to include "Mar 24"
+    end
+
+    it "displays correct number of posts from past week" do
+      create(:article, published_at: Time.zone.local(2022, 3, 30))
+      create(:article, published_at: Time.zone.local(2022, 3, 29))
+      create(:article, published_at: Time.zone.local(2022, 3, 24))
+      create(:article, published_at: Time.zone.local(2022, 3, 10))
+      page.evaluate_script("window.location.reload()")
+
+      expect(page.body).to include "2</span> Posts"
+    end
+
+    it "does not display posts from previous week" do
+      create(:article, published_at: Time.zone.local(2022, 3, 10))
+      page.evaluate_script("window.location.reload()")
```

Comment:
Since the post counts for the timeframe are rendered on the backend, and present in the html response, these test cases would work fine in a request spec (like spec/requests/admin/overview_spec.rb) rather than needing to use headless chrome to load the page. That's faster than a js enabled system test, and possibly closes another edge case.

I think the best reason to avoid putting this in a system test is that the timezone randomization we use (every time you run the specs you're assigned a random time zone) may only affect the rails server (times are modified to the server's local time), but the headless chrome process may be running on your system time, in CI that's UTC, but my computer or yours may differ. 

When the system time of chrome, and the rails server time set randomly during test execution, are more than 12 hours apart there appears to be a failure. For example, I'm UTC-5, running this test with `ZONEBIE_TZ="Australia/Perth"` consistently fails because the dates in the browser change relative to the backend (dates are localized by the browser) - my time shows "Mar 25 to Yesterday" rather than "Mar 24 to Yesterday". 

It's more clear that the headless browser is not respecting the timezone change here if you set the date farther away from today 

```
  describe "analytics" do
    before do
      travel_to Time.zone.local(2022, 4, 30)
    end

    it "starts a week before" do
      expect(page.body).to include "Apr 24"
    end
```

This fails showing March dates (my current last week) in the rendered postsChart, while the response was served as though it were April.

https://gist.github.com/djuber/c80ce1de406677336010804fb63eb52e this is basically moving the tests out of the system test to test the html only.

Reply:
That makes sense, I was wondering whether this was the best place for these specs. But I do think adding _only_ request specs might not fully cover everything, because we're relying on the `charts_data` working properly too. I'll add some more examples in there too, to make sure we've covered everything. Thanks for all your feedback, @djuber!